inloop/solutions/views.py

```diff
@@ -104,10 +104,18 @@ def get(self, request, **kwargs):
             junit.xml_to_dict(xml) for xml in xml_reports
         ]
 
+        upfiles = []
+        fp = Solution.objects.filter(id=solution.id)[0]
+        fileNames = SolutionFile.objects.filter(solution=fp.id)
+        for fn in fileNames:
+            file = open(str(fp.path) + "/" + str(fn), "r")
```

Comment:
* Where is this file closed? You should always use `open` with a context manager and specify the encoding.
* You can iterate over the `SolutionFile`s of a `Solution` using the related set `self.solutionfile_set`.
* Each `SolutionFile` has a `relative_path` and `absolute_path` property, which is a `pathlib.Path`, which in turn provides `Path.open()`.

Together:

```python
for solutionfile in self.solutionfile_set.all():
    with solutionfile.absolute_path.open(encoding="utf-8") as f:
        upfiles.append({"code": f.read(), "title": solutionfile.name)
```